packages/event-handler/src/rest/BaseRouter.ts

```diff
@@ -76,6 +105,68 @@ abstract class BaseRouter {
     }
   }
 
+  /**
+   * Handles errors by finding a registered error handler or falling
+   * back to a default handler.
+   *
+   * @param error - The error to handle
+   * @returns A Response object with appropriate status code and error details
+   */
+  protected handleError(error: Error): Response {
+    const handler = this.errorHandlerRegistry.resolve(error);
+    if (handler != null) {
+      try {
+        const body = handler(error);
```

Comment:
Should we `await` this?

Reply:
Yeah, I wasn't sure whether to make the error handlers async. I guess there's definitely cases where user might want to, e.g., do something that involves a network call in their error handlers so we probably should.

Reply:
Yeah, makes sense!